syweb/webclient/app/components/matrix/matrix-service.js

```diff
@@ -243,118 +273,161 @@ function($http, $window, $timeout, $q) {
             // registration is composed of multiple requests, to check you can
             // register, then to actually register. This deferred will fire when
             // all the requests are done, along with the final response.
             var deferred = $q.defer();
             var path = "/register";
-            
-            // check we can actually register with this HS.
-            doRequest("GET", path, undefined, undefined).then(
-                function(response) {
-                    console.log("/register [1] : "+JSON.stringify(response));
-                    var flows = response.data.flows;
-                    var knownTypes = [
-                        "m.login.password",
-                        "m.login.recaptcha",
-                        "m.login.email.identity"
-                    ];
-                    // if they entered 3pid creds, we want to use a flow which uses it.
-                    var useThreePidFlow = threepidCreds != undefined;
-                    var flowIndex = 0;
-                    var firstRegType = undefined;
-                    
-                    for (var i=0; i<flows.length; i++) {
-                        var isThreePidFlow = false;
-                        if (flows[i].stages) {
-                            for (var j=0; j<flows[i].stages.length; j++) {
-                                var regType = flows[i].stages[j];
-                                if (knownTypes.indexOf(regType) === -1) {
-                                    deferred.reject("Unknown type: "+regType);
-                                    return;
-                                }
-                                if (regType == "m.login.email.identity") {
-                                    isThreePidFlow = true;
-                                }
-                                if (!useCaptcha && regType == "m.login.recaptcha") {
-                                    console.error("Web client setup to not use captcha, but HS demands a captcha.");
-                                    deferred.reject({
-                                        data: {
-                                            errcode: "M_CAPTCHA_NEEDED",
-                                            error: "Home server requires a captcha."
-                                        }
-                                    });
-                                    return;
-                                }
-                            }
-                        }
-                        
-                        if ( (isThreePidFlow && useThreePidFlow) || (!isThreePidFlow && !useThreePidFlow) ) {
-                            flowIndex = i;
-                        }
-                        
-                        if (knownTypes.indexOf(flows[i].type) == -1) {
-                            deferred.reject("Unknown type: "+flows[i].type);
-                            return;
+            if (bind_email === undefined) bind_email = false;
```

Comment:
The SDK doesn't tend to inline like this because it is extremely error prone when you come to modify it later on. Please use `{ }` or use ternary operators.